docs/sdks/python/knowledge-bases.mdx

```diff
@@ -0,0 +1,1069 @@
+| order_id | product               | notes                  |
++----------+-----------------------+------------------------+
+| A1B      | Wireless Mouse        | Request color: black   |
+| 3XZ      | Bluetooth Speaker     | Gift wrap requested    |
+| Q7P      | Aluminum Laptop Stand | Prefer aluminum finish |
++----------+-----------------------+------------------------+
+```
+
+<Tip>
+Go to the *Complete Example* section below to find out how to access this sample data.
+</Tip>
+
+Here is how to propagate this change into the knowledge base.
+
+```python
+my_kb.insert_query(
+    server.databases.sample_data.tables.orders.filter(order_id='Q7P')
+)
+```
+
+The knowledge base matches the ID value to the existing one and updates the data if required.
+
+### Insert Data using Partitions
+
+TODO: os this available in python sdk?
+
+In order to optimize the performance of data insertion into the knowledge base, users can set up partitions and threads to insert batches of data in parallel. This also enables tracking the progress of data insertion process including cancelling and resuming it if required.
+
+Here is an example.
+
+```python
+project.query(
+    '''
+    INSERT INTO my_kb
+    SELECT order_id, product, notes
+    FROM sample_data.orders
+    USING
+        batch_size = 200,
+        track_column = order_id,
+        threads = 10,
+        error = 'skip';
+    '''
+)
+```
+
+The parameters include the following:
+
+* `batch_size` defines the number of rows fetched per iteration to optimize data extraction from the source. It defaults to 1000.
+
+* `threads` defines threads for running partitions. Note that if the [ML task queue](/setup/custom-config#overview-of-config-parameters) is enabled, threads are used automatically. The available values for include a number of threads to be used, for example, `threads = 10`, or a boolean value that defines whether to enable threads, setting `threads = true`, or disable threads, setting `threads = false`.
+
+* `track_column` defines the column used for sorting data before partitioning.
+
+* `error` defines the error processing options. The available values include `raise`, used to raise errors as they come, or `skip`, used to subside errors. It defaults to `raise` if not provided.
+
+After executing the `INSERT INTO` statement with the above parameters, users can view the data insertion progress by querying the `information_schema.queries` table.
+
+```python
+project.query(
+    '''
+    SELECT * FROM information_schema.queries;
+    '''
+)
+```
+
+Users can cancel the data insertion process using the process ID from the `information_schema.queries` table.
+
+```python
+project.query(
+    '''
+    SELECT query_cancel(1);
+    '''
+)
+```
+
+Note that canceling the query will not remove the already inserted data.
+
+Users can resume the data insertion process using the process ID from the `information_schema.queries` table.
+
+```python
+project.query(
+    '''
+    SELECT query_resume(1);
```

Comment:
Maybe we have short section to explain the query_resume and query_cancel functions?

Reply:
added short sentences below